WPCOM Plugins: add plugin panel styles

drw158 opened the pull request:
To test, go to http://calypso.localhost:3000/plugins/test.wordpress.com

This PR updates the design of the wpcom Plugins view. **This view is not launched, it is hidden.**

- small viewports get a 1 column grid
- `View all standard plugins` link goes to another view with all standard plugins, similar to http://calypso.localhost:3000/plugins/browse/popular/test.wordpress.com

To do in another PR:

- Link up actions for the Purchase buttons
- Add the `View all standard plugins` view
- Change gridicon for each plugin

Before:
![screen shot 2016-03-25 at 5 33 03 pm](https://cloud.githubusercontent.com/assets/618551/14055415/c254dfa2-f2af-11e5-80bd-0f087f2e7c48.png)

After:
![screen shot 2016-03-28 at 4 09 21 pm](https://cloud.githubusercontent.com/assets/618551/14090373/95b956b0-f4ff-11e5-9a85-ccaa18d217cb.png)

dmsnell commented:
@drw158 you're amazing. thanks for working on this. I think there's something funny with the formatting, but I will try and take a look at that and commit if need be. Is there a reason that the `actionButton`s were left in?

drw158 commented:
Thanks! That'd be great. There is some cleanup to be done. There is no reason why I left that in there. Probably some funky JS indenting too.

drw158 commented:
@dmsnell 

I just updated the design again. I think it's almost ready for review. 

When you get a minute can you remove the JS that's not needed anymore?

- I removed the foldable card component
- New buttons ( `actionButton` is not needed)

dmsnell commented on client/my-sites/plugins-wpcom/info-header.jsx:
```diff
@@ -1,23 +1,19 @@
+			>
+				<NoticeAction href="https://en.support.wordpress.com/plugins/">
+					{ "Learn More" }
+				</NoticeAction>
```
@drw158 this opens up in the current browser window and I was wondering if it should open in a new tab

drw158 replied:
Oh yea, new tab would be better.

apeatling commented:
@drw158: Couple of questions on the design:

* Will we be looking at getting different icons for the plugins? At the very least, what about using different "plug" icons with various background colors? We have so much blue in Calypso already.
* The text in the plugins boxes seem darker than in other parts of Calypso, is this me seeing things?
* What will happen when I hit the "View all standard plugins" link?

drw158 commented:
> Will we be looking at getting different icons for the plugins?

The plan right now is to assign each plugin a different Gridicon. At least assign each "category" of plugin to a certain icon.

All the dark text is using our $gray-dark color.

On my first wireframes, I was imagining the list to expand inline on the page, but after this PR I decided that it should probably function like the "See All" link on the add plugins page (header cake on the next view). As we add more plugins, I think it would become unwieldy to display so many inline.

wordpress.com/plugins/browse/testblog.com

![screen shot 2016-03-29 at 2 42 26 pm](https://cloud.githubusercontent.com/assets/618551/14121308/7abb5e72-f5bc-11e5-9c78-7ff062632fbc.png)
